File: BACKEND/Application/Groups/Responses/BaseGroupResponse.cs
namespace Application.Groups.Responses
{
    public class BaseGroupResponse
    {
        public Guid Id { get; set; }
        public required string CreatorName { get; set; }
        public required string Name { get; set; }
        public required string Description { get; set; }

        public required string Visibility { get; set; }
        public required string JoinPolicy { get; set; }
        public required string SizePreset { get; set; }

        public int MaxMembers { get; set; }
        public int MaxModerators { get; set; }

        public DateTimeOffset CreatedAt { get; set; }
    }
}

File: FRONTEND/backgammon-frontend/src/app/features/groups/models/api/responses/base-group.response.ts
import { GroupVisibility } from '../../enums/group-visibility.type';
import { GroupJoinPolicy } from '../../enums/group-join-policy.type';
import { GroupSizePreset } from '../../enums/group-size-preset.type';
import { GroupUserState } from '../../enums/group-user-state.type';

export interface BaseGroupResponse {
  id: string;
  creatorName: string;
  name: string;
  description: string;

  visibility: GroupVisibility;
  joinPolicy: GroupJoinPolicy;
  sizePreset: GroupSizePreset;

  maxMembers: number;
  maxModerators: number;

  groupUserState: GroupUserState | null;

  createdAt: string;
}
